runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -99,6 +102,60 @@ GC_ClassLoaderClassesIterator::switchToSystemMode()
 	return isSystemClassLoader;
 }
 
+J9Class *
+GC_ClassLoaderClassesIterator::nextArrayClass()
+{
+	while (_arrayState != STATE_DONE) {
+		switch (_arrayState) {
+		case STATE_VALUETYPEARRAY:
+			{
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+				J9Class *nullRestrictedArray = J9CLASS_GET_NULLRESTRICTED_ARRAY(_startingClass);
+				if (NULL != nullRestrictedArray) {
+					_iterateArrayClazz = nullRestrictedArray;
+					_arrayState = STATE_VALUETYPEARRAYLIST;
+				} else
+#endif /* defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES) */
+				{
+					_arrayState = STATE_ARRAY;
+				}
+			}
+			break;
+		case STATE_VALUETYPEARRAYLIST:
+			if (NULL != _iterateArrayClazz) {
+				_iterateArrayClazz = _iterateArrayClazz->arrayClass;
+			} else {
+				_arrayState = STATE_ARRAY;
+			}
+			break;
+		case STATE_ARRAY:
+			if (NULL != _startingClass->arrayClass) {
+				_iterateArrayClazz = _startingClass->arrayClass;
+				_arrayState = STATE_ARRAYLIST;
+			} else {
+				_iterateArrayClazz = NULL;
+				_arrayState = STATE_DONE;
+			}
+			break;
+		case STATE_ARRAYLIST:
+			if (NULL != _iterateArrayClazz) {
+				_iterateArrayClazz = _iterateArrayClazz->arrayClass;
+			} else {
+				_iterateArrayClazz = NULL;
```

Comment:
Please remove this redundant assignment (notice line 141).

Reply:
I'm running another Java 8 build with this update for good measure http://vmfarm.rtp.raleigh.ibm.com/build_info.php?build_id=77729